lib/command/Command.js

```diff
@@ -279,17 +279,29 @@ class Command {
         return true;
     }
 
-    process(args, msg) {
+    async process(args, msg) {
         var shouldDelete = this.deleteCommand && msg.channel.guild && msg.channel.permissionsOf(msg._client.user.id).has("manageMessages");
 
         if(this.hooks.preCommand) {
-            this.hooks.preCommand(msg, args);
+            if (this.hooks.preCommand(msg, args) !== undefined) {
+                const response = await this.hooks.preCommand(msg, args);
+                msg = response.msg ? response.msg : msg;
+                args = response.args ? response.args : args;    
+            } else {
+                this.hooks.preCommand(msg, args);
```

Comment:
The `else` block here should be removed, it results in the hook being run twice if it doesn't return anything